src/components/ColorInput/index.js

```diff
@@ -0,0 +1,372 @@
+    StyledCard,
+    StyledContent,
+    StyledInput,
+    ColorInputContainer,
+} from './styled';
+import {
+    useOutsideClick,
+    useWindowResize,
+    useUniqueIdentifier,
+    useReduxForm,
+    useErrorMessageId,
+    useLabelId,
+} from '../../libs/hooks';
+import {
+    StyledFlagContainer,
+    StyledIndicator,
+    StyledInputContainer,
+    StyledTrigger,
+} from '../PhoneInput/styled';
+import AssistiveText from '../AssistiveText';
+import RenderIf from '../RenderIf';
+import HelpText from '../Input/styled/helpText';
+import ErrorText from '../Input/styled/errorText';
+import { useFocusIndex, useHandleBlur, useHandleFocus } from '../PhoneInput/hooks';
+import ColorSample from './colorSample';
+import { decomposeColor, hexToRgb, hexToRgba, recomposeColor } from '../../styles/helpers/color';
+import isHexColor from '../../styles/helpers/color/isHexColor';
+import StyledIconContainer from '../Input/styled/iconContainer';
+import getHexString from './helpers/getHexString';
+import getColorValue from './helpers/getColorValue';
+
+/**
+ * Provides a color input with an improved color picker.
+ * @category Form
+ */
+const ColorInput = props => {
+    const {
+        id,
+        className,
+        style,
+        value,
+        onChange,
+        defaultColors,
+        variant,
+        onClick,
+        label,
+        labelAlignment,
+        hideLabel,
+        required,
+        readOnly,
+        disabled,
+        error,
+        tabIndex,
+        name,
+        placeholder,
+        bottomHelpText,
+        onFocus,
+        onBlur: blurInProps,
+    } = useReduxForm(props);
+    const [isOpen, setIsOpen] = useState(false);
+    const [colorValue, setColorValue] = useState(getColorValue(value));
+    const [sampleColor, setSampleColor] = useState(value);
+    const pickerRef = useRef();
+    const containerRef = useRef();
+    const triggerRef = useRef();
+    const alphaInputRef = useRef();
+    const inputRef = useRef();
+
+    const inputId = useUniqueIdentifier('color-input');
+    const errorMessageId = useErrorMessageId(error);
+    const labelId = useLabelId(label);
+
+    const [focusIndex, setFocusIndex] = useFocusIndex(
+        containerRef,
+        triggerRef,
+        inputRef,
+        alphaInputRef,
+    );
+
+    useOutsideClick(
+        containerRef,
+        () => {
+            setFocusIndex(-1);
+        },
+        focusIndex > -1,
+    );
+    useOutsideClick(
+        pickerRef,
+        event => {
+            if (
+                event.target !== triggerRef.current &&
+                !triggerRef.current.contains(event.target) &&
+                !pickerRef.current.contains(event.target)
+            ) {
+                setIsOpen(false);
+            }
+        },
+        isOpen,
+    );
+    useWindowResize(() => setIsOpen(false), isOpen);
+
+    const onBlur = event => {
+        const { hex } = value;
+        if (!isHexColor(hex)) setSampleColor(undefined);
+        blurInProps(event);
```

Comment:
onBlur and onFocus props should send the value, the same as in the onChange, that is because when you use this component in a form the form state can change with the onBlur or the onFocus then we must send the same data